R/purityA-frag4feature.R

```diff
@@ -225,9 +233,9 @@ fsub2  <- function(pro, allpeaks, intense, ppm, fullp=FALSE, use_grped=FALSE){
   }
 
   if (fullp){
```

Comment:
Perhaps we can prevent converting to 0. And just skip the NA values in the filter stage e.g.

```
  if (fullp){
    rtmin_col <- "rtmin_full"
    rtmax_col <- "rtmax_full"
  }else{
    rtmin_col <- "rtmin"
    rtmax_col <- "rtmax"
  }

  mtchRT <- allpeaks[prt>=allpeaks[,rtmin_col] &
                     prt<=allpeaks[,rtmax_col] &
                     !is.na(allpeaks[,rtmin_col]) &
                     !is.na(allpeaks[,rtmax_col]),]
```

Reply:
I thought the NAs are given from XCMS for the "filled" peaks? So I was just trying to preserve that information. 

My understanding is that if the "filled" peak is not peak picked (e.g. from centWave) certain columns cannot be calculated as so are given an NA value.